src/codegen/general.rs

```diff
@@ -149,6 +149,48 @@ pub fn define_boxed_type(
     Ok(())
 }
 
+pub fn define_auto_boxed_type(
+    w: &mut Write,
+    type_name: &str,
+    glib_name: &str,
+    get_type_fn: &Option<String>,
+) -> Result<()> {
+    let get_type_fn = if let Some(ref get_type_fn) = *get_type_fn {
+        get_type_fn
+    } else {
+        panic!(
+            "Record {} has record_boxed=true but don't have glib:get_type function",
+            type_name
+        );
+    };
+
+    try!(writeln!(w, ""));
+    try!(writeln!(w, "glib_wrapper! {{"));
+    try!(writeln!(
+        w,
+        "\tpub struct {}(Boxed<ffi::{}>);",
+        type_name,
+        glib_name
+    ));
+    try!(writeln!(w, ""));
+    try!(writeln!(w, "\tmatch fn {{"));
```

Comment:
We have tabulations now? :O

Reply:
And long long time ;)

Reply:
Oh damn. I'm so late!

Reply:
It replaced with 4 spaces by https://github.com/gtk-rs/gir/blob/master/src/writer/untabber.rs

Reply:
Just saw it a few minutes ago (I was very surprised). Getting old, my memory can't follow up...